### ISSUE TITLE
Reinstate support for filtering and arranging lubridate Periods and Intervals

### ISSUE THREAD

**Issue by richierocks** [tidyverse/dplyr#3695]
Support for `filter()`ing and `arrange()`ing columns of class `Period` and `Interval` was [recently dropped](https://community.rstudio.com/t/am-i-crazy-having-issues-with-dplyr-filter/7710) from `dplyr`. The ability to work with these types is useful, so I'd like the feature reinstating.

(By 'useful', I mean in particular that there are DataCamp courses depending upon this feature. For example, [Working with Dates and Times in R](https://campus.datacamp.com/courses/working-with-dates-and-times-in-r/arithmetic-with-dates-and-times?ex=11).)

---

To reproduce:

```r
library(dplyr)                        
library(lubridate)                    
x <- data_frame(                      
  start = as.Date("2000-01-01") + 0:19, 
  finish = as.Date("2001-01-01") + 19:0,
  interval = start %--% finish          
)                                     
x %>%                                 
  arrange(interval)                     
#> Error in arrange_impl(.data, dots): Column `interval` classes Period and Interval from lubridate are currently not supported.
                                      
x %>%                                 
  filter(interval > 0)                  
#> Error in filter_impl(.data, quo): Column `interval` classes Period and Interval from lubridate are currently not supported.
```

**Comment by krlmlr** [comment]
Are you sure these features worked as expected? How is comparison defined between intervals?

We should definitely introduce proper support for these and other data types, #2432 is the canonical issue.

**Comment by richierocks** [comment]
@krlmlr Comparison for intervals was on the length of the interval (in seconds) &ndash; I think this is reasonable. You can use `unclass(lubridate::int_length(x))`.

**Comment by romainfrancois** [comment]
There are a few things we need to figure out before we can make these things work again. Some code will need to be factored out of dplyr or reimplemented so that 📦 like lubridate can implement a method that does the right thing without necessarily depend on dplyr. 

This is probably for the 0.9.0 horizon at least

**Comment by hadley** [comment]
`filter()` should work in 0.8.0, `arrange()` will need to wait for 0.9 & vctrs support

**Comment by hadley** [comment]
Duplicate of #2432
